From 8b8995f0f4bf5d9b96b98af8c4e0ed26f4944b7b Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Thu, 24 Apr 2014 22:51:54 +0200
Subject: [PATCH 16/17] MINOR: stats: always emit HTTP/1.1 in responses

We used to emit either 1.0 or 1.1 depending on whether we were sending
chunks or not. This condition is useless, better always send 1.1. Also
that way at least clients and intermediary proxies know we speak 1.1.
The "Connection: close" header is still set anyway.
---
 src/dumpstats.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/dumpstats.c b/src/dumpstats.c
index dd017e1..5f28e1c 100644
--- a/src/dumpstats.c
+++ b/src/dumpstats.c
@@ -4357,11 +4357,10 @@ static int stats_send_http_headers(struct stream_interface *si)
 	struct appctx *appctx = objt_appctx(si->end);
 
 	chunk_printf(&trash,
-		     "HTTP/1.%c 200 OK\r\n"
+		     "HTTP/1.1 200 OK\r\n"
 		     "Cache-Control: no-cache\r\n"
 		     "Connection: close\r\n"
 		     "Content-Type: %s\r\n",
-		     (appctx->ctx.stats.flags & STAT_CHUNKED) ? '1' : '0',
 		     (appctx->ctx.stats.flags & STAT_FMT_HTML) ? "text/html" : "text/plain");
 
 	if (uri->refresh > 0 && !(appctx->ctx.stats.flags & STAT_NO_REFRESH))
-- 
1.8.3.2

